pgjdbc/src/main/java/com/yugabyte/ysql/LoadBalancer.java

```diff
@@ -0,0 +1,35 @@
+package com.yugabyte.ysql;
+
+import java.util.List;
+import java.util.Map;
+
+/**
+ * An interface for any load balancing policy to implement. The LoadBalanceManager invokes the
+ * implemented methods while processing a new connection request.
+ *
+ * @see LoadBalanceManager
+ * @see ClusterAwareLoadBalancer
+ * @see TopologyAwareLoadBalancer
+ */
+public interface LoadBalancer {
+
+  /**
+   * @param e The {@link com.yugabyte.ysql.LoadBalanceManager.NodeInfo} object for the host
+   * @return true, if a host is eligible to be considered for a connection request
+   */
+  boolean isHostEligible(Map.Entry<String, LoadBalanceManager.NodeInfo> e);
```

Comment:
I think the host is eligible or not can be found out by the loadbalancer itself. Not sure why we need a public api for this.

Reply:
The loadbalancer needs the `NodeInfo` to determine if it's eligible and the `NodeInfo` is maintained by the `LoadBalanceService`. So added this method to the interface instead of passing the list of `NodeInfo`s to the loadbalancer.